src/pages/index.js

```diff
@@ -25,26 +36,109 @@ const cardData = {
+      job: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((error) => {
+    console.error("Error setting userInfo:", error);
+  });
+
+/* ------------------------------------------------- */
+/*                     Form Validation
+/* ------------------------------------------------- */
+const formValidators = {};
+
+const enableValidation = (settings) => {
+  const formList = Array.from(document.querySelectorAll(settings.formSelector));
+  formList.forEach((formElement) => {
+    const validator = new FormValidator(settings, formElement);
+    const formName = formElement.getAttribute("name");
+
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+enableValidation(settings);
+
 // /* ------------------------------------------------- */
-// /*                 Profile Edit Form
+// /*                 Profile Avatar
 /* ------------------------------------------------- */
+const handleAvatarFormSubmit = (formData) => {
+  const newAvatarUrl = formData.avatarUrl;
 
+  return api.updateAvatar(newAvatarUrl).then((userData) => {
+    console.log("Avatar updated on server:", userData.avatar);
+    userInfo.setUserAvatar(userData.avatar);
+  });
+};
+
+const profileEditAvatar = new PopupWithForm(
+  "#update-avatar-modal",
+  handleAvatarFormSubmit,
+  (formValues) => {
+    profileEditAvatar.renderLoading(true);
+    const avatarUrl = formValues.avatar;
+    api
+      .updateAvatar(avatarUrl)
+      .then((userData) => {
+        console.log("Avatar URL:", userData.avatar);
+        userInfo.setUserAvatar(userData.avatar);
+        profileEditAvatar.close();
+      })
+      .catch((err) => {
+        console.error(`Error Submitting Form: ${err}`);
+      });
+  }
+);
+
+const profileEditAvatarButton = document.querySelector(".profile__edit-icon");
+
+profileEditAvatar.setEventListeners();
+
+profileEditAvatarButton.addEventListener("click", () => {
+  profileEditAvatar.open();
+});
+
+// /* ------------------------------------------------- */
+// /*                 Profile Edit
+/* ------------------------------------------------- */
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   (formValues) => {
-    userInfo.setUserInfo({
-      name: formValues.title,
-      job: formValues.description,
-    });
+    profileEditPopup.renderLoading(true);
+    return api
+      .setUserInfo({
+        name: formValues.title,
+        about: formValues.description,
+      })
+      .then((data) => {
+        userInfo.setUserAvatar(data.avatar);
 
-    profileEditPopup.close();
-    const formName = profileEditPopup.getForm().getAttribute("name");
-    const validator = formValidators[formName];
-    validator.disableButton();
+        const formName = profileEditPopup.getForm().getAttribute("name");
+        const validator = formValidators[formName];
+        validator.disableButton();
+        profileEditPopup.close();
+      })
+      .catch((error) => {
```

Comment:
`catch` is not needed here because you **catch** errors inside `PopupWithForm`

**Please, correct it everywhere in the code**